jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1923,6 +1924,13 @@ ecma_builtin_typedarray_prototype_dispatch_routine (uint8_t builtin_routine_id,
     }
     case ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_BYTELENGTH_GETTER:
     {
+      ecma_object_t *buffer_p = ecma_typedarray_get_arraybuffer (typedarray_p);
+
+      if (ecma_arraybuffer_is_detached (buffer_p))
+      {
+        return ecma_make_uint32_value (0);
+      }
+
       uint32_t length = ecma_typedarray_get_length (typedarray_p);
```

Comment:
What about `ecma_typedarray_get_length ` returning with 0 for detached buffers. This could fix this case and the case below.